### PR TITLE
[fix]: [PIE-8390]: Fix issue where all events were getting marked success

### PR DESCRIPTION
**Pull request by namanvermah**
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44095)

### CODE REVIEW

File: pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java [hunk 1]
```diff
@@ -88,7 +88,8 @@ void evaluatePoliciesAndUpdateResult(PipelineEntity pipelineEntity, TemplateMerg
     ValidationResult governanceValidationResult = templateValidationResult.withGovernanceMetadata(governanceMetadata);
     if (governanceMetadata.getDeny()) {
       validationService.updateEvent(validationEvent.getUuid(), ValidationStatus.FAILURE, governanceValidationResult);
+    } else {
+      validationService.updateEvent(validationEvent.getUuid(), ValidationStatus.SUCCESS, governanceValidationResult);
```
[the remark this answers is not in the document]
Comment (namanvermah) [reply pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java:92]: yes adding, that's why not put it on slack yet

[the remark this answers is not in the document]
Comment (namanvermah) [reply pipeline-service/service/src/main/java/io/harness/pms/pipeline/validation/async/handler/PipelineAsyncValidationHandler.java:92]: Done

**Review by naidu-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44095)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by ankittiwari-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44095)* status: :shipit: complete! ✓

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44095)* status: :shipit: complete! ✓

**Pull request merged**